src/legacy.js

```diff
@@ -0,0 +1,59 @@
+
+/**
+ * @typedef {Object} Options
+ * @prop {number} [damping = 0.1] - Levenberg-Marquardt lambda parameter
+ * @prop {number} [dampingDrop = 0.1] - The constant used to lower the damping parameter
+ * @prop {number} [dampingBoost = 1.5] - The constant used to increase the damping parameter
+ * @prop {number} [maxDamping] - Maximum value for the damping parameter
+ * @prop {number} [minDamping] - Minimum value for the damping parameter
+ * @prop {number} [gradientDifference = 1e-6] - The "infinitesimal" value used to approximate the gradient of the parameter space
+ * @prop {Array<number>} [initialValues] - Array of initial parameter values
+ * @prop {number} [maxIterations = 100] - Maximum of allowed iterations
+ * @prop {number} [residualEpsilon = 1e-6] - Minimum change of the sum of residuals per step – if the sum of residuals changes less than this number, the algorithm will stop
+ * @prop {Array<number>} [maxValues] - Maximum values for the parameters
+ * @prop {Array<number>} [minValues] - Minimum values for the parameters
+ * @prop {{rough: number, fine: number}} [errorPropagation] - How many evaluations (per point per step) of the fitted function to use to approximate the error propagation through it
```

Comment:
Where are these `rough` and `fine` values used?

Reply:
Yes, I apparently forgot to remove them. In the best-case scenario, the algorithm would interpolate between those two as it approached the minimum, but I didn't manage to do it reliably, so it's best to remove those options.